File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/Codes1.1/CFD2D/IsentropicVortexBC2D.m
function Q = IsentropicVortexBC2D(xin, yin, nxin, nyin, mapI, mapO, mapW, mapC, Q, time);
  
%  function [Q] = IsentropicVortexBC2D(xin, yin, nxin, nyin, mapI, mapO, mapW, mapC, Q, time);
% Purpose: Impose boundary conditions on 2D Euler equations on weak form

Qbc = IsentropicVortexIC2D(xin, yin, time);

mapB = [mapI;mapO;mapW];

for n=1:4
  Qn = Q(:,:,n);  Qbcn = Qbc(:,:,n); Qn(mapB) = Qbcn(mapB);
  Q(:,:,n) = Qn;
end
return

File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/nudg++/trunk/Src/Examples2D/CurvedEuler2D/IsentropicVortexBC2D.cpp
// IsentropicVortexBC2D.m
// function [Q] = IsentropicVortexBC2D(Q);
// 2007/06/30
//---------------------------------------------------------
#include "NDGLib_headers.h"
#include "CurvedEuler2D.h"


//---------------------------------------------------------
void CurvedEuler2D::IsentropicVortexBC2D
(
  const DVec& xi, 
  const DVec& yi, 
  const DVec& nxi, 
  const DVec& nyi, 
  const IVec& tmapI, 
  const IVec& tmapO, 
  const IVec& tmapW, 
  const IVec& tmapC, 
        double ti, 
        DMat& Qio
)
//---------------------------------------------------------
{
  // function [Q] = Euler2Dbc(Q)
  // Purpose: Impose boundary conditions on 2D Euler equations on weak form

  // Example is static isentropic vortex

  static DMat Qbc;  Qbc.resize(Qio);  DVec qn, qbcn;
  
  // Load Qbc with IsentropicVortex2D for time=ti
  IsentropicVortexIC2D(xi, yi, ti, Qbc);

  // map isentropic vortex values onto boundary nodes
  // Note: this loop updates Qio(:,n) in-place:
  int Nr=Qio.num_rows();
  for (int n=1; n<=4; ++n) {
    qn.borrow  (Nr, Qio.pCol(n));   // qn -> Q(:,n)
    qbcn.borrow(Nr, Qbc.pCol(n));
    qn(gmapB) = qbcn(gmapB);
  }
}
